web/components/chat/ChatContext.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  ReactNode,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { DBMessage } from './types';
+import { createClient } from '@/utils/supabase/client';
+import { listUserConversations } from '@utils/lib/messaging/services';
+
+type ChatContextProps = {
+  newMessages: Record<string, DBMessage[]>;
+};
+
+const ChatContext = createContext<ChatContextProps | null>(null);
+
+export const ChatProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const supabase = createClient();
+  const [newMessages, setNewMessages] = useState<Record<string, DBMessage[]>>(
+    {}
+  ); // conversationId: [message, message, ...]
+  const [conversationsIds, setConversationsIds] = useState<string[]>([]);
+
+  useEffect(() => {
+    // fetch the conversations of the user to get the ids
+    async function fetchConversations() {
+      const { data: conversationsFetched, error: conversationsError } =
+        await listUserConversations(supabase);
+
+      if (conversationsError) {
+        throw new Error("Couldn't fetch conversations");
+      }
+
+      setConversationsIds(
+        conversationsFetched.map((conversation) => conversation.id)
+      );
+    }
+
+    fetchConversations();
+  }, [supabase]);
+
+  useEffect(() => {
+    // Listen for new messages inserted in the database
+    const changes = supabase
+      .channel('messages')
+      .on(
+        'postgres_changes',
+        {
+          event: 'INSERT',
+          schema: 'public',
+          table: 'message',
```

Comment:
Ici tu peux ajouter un filter pour ecouter que les conversations souhaitées 
`filter: 'conversation_id=in.(id_1, id_2, etc)',`

Reply:
parce que la tu ecoutes les messages de tous le monde